Source/Core/DolphinQt2/Config/Mapping/MappingWidget.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QPushButton>
+
+#include "DolphinQt2/Config/Mapping/MappingWindow.h"
+#include "InputCommon/ControllerEmu/Control/Control.h"
+#include "InputCommon/ControllerEmu/ControlGroup/ControlGroup.h"
+
+#include "DolphinQt2/Config/Mapping/MappingWidget.h"
+
+MappingWidget::MappingWidget(MappingWindow* window) : m_parent(window)
+{
+  connect(window, &MappingWindow::LoadProfile, this, &MappingWidget::OnLoadProfile);
+  connect(window, &MappingWindow::SaveProfile, this, &MappingWidget::OnSaveProfile);
+}
+
+MappingWindow* MappingWidget::GetParent() const
+{
+  return m_parent;
+}
+
+int MappingWidget::GetPort() const
+{
+  return m_parent->GetPort();
+}
+
+QGroupBox* MappingWidget::CreateGroupBox(const char* name, ControllerEmu::ControlGroup* group) const
+{
+  QGroupBox* group_box = new QGroupBox(tr(name));
```

Comment:
Are you certain that Qt Linguist would be able to pick up all the strings passed to the function this way? Qt's documentation on localizing usually wraps those strings in QT_TR_NOOP (and doesn't mention its behavior on variables, unless I missed that). In that case it would probably be better to just have the function take a const reference to a QString and put `tr()` around the strings at the call site.